android-components/components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -860,6 +860,18 @@ sealed class CookieBannerAction : BrowserAction() {
         CookieBannerAction()
 }
 
+/**
+ * [BrowserAction] implementations related to updating the [SessionState.ShoppingProduct] of a single [SessionState] inside
+ * [BrowserState]
+ */
+sealed class ShoppingProductAction : BrowserAction() {
```

Comment:
Could this be internal?

Reply:
None of the other actions here are. Why do you suggest it should be internal?

Reply:
As it's only used in the `browser/state` module, it can be internal and we don't need to expose this to Fenix or other consumers.

Reply:
Rahul, I followed the convention already established and wrote this class in exactly the same way as other classes are written. I think if the team that takes care of this repository now wants to change the conventions - that needs to be a separate bug. This is out of scope of this patch.